### PR TITLE
Lint More

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This adds in a bunch of lint rules that were previously disabled along with all the small little code changes to allow these rules to run without errors or warnings

## Why?
Because we were talking about linting for a11y stuff and I agree'd to reintroduce those rules to our codebase. In the end I don't think these add a great deal to help with accesibility and we get more from raising awareness /; writing semantic code but once I started in with these rules I added a bunch more

## What's New?
I added these rules:
`no-named-default`
`no-unresolved`
`aria-role`
`click-events-have-key-events`
`img-redundant-alt`
`label-has-associated-control`
`no-noninteractive-element-to-interactive-role`
`no-noninteractive-tabindex`
`no-redundant-roles`
`no-cycle`
`no-use-before-define`
`no-unused-vars`
`no-duplicate-case`
`no-return-assign`
`jsx-props-no-spreading` 🙏 
`no-unescaped-entities`

## What's Old?
I removed linting for `no-explicit-any` - because I just didn't fancy writing loads of ignore statements and we should know better anyway and `no-danger` because, lets face it, we like a bit of danger

## Link to supporting Trello card
https://trello.com/c/8saHIavq/1306-lint-more

### CODE REVIEW

File: src/app/logging.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ const logFields = (logEvent: any): any => {
     };
 };
 
-addLayout('json', config => {
+addLayout('json', () => {
```
Comment (gtrufitt) [inline src/app/logging.ts]: How did this get through?!

File: src/web/components/Nav/ExpandedMenu/ExpandedMenuToggle/VeggieBurger.tsx [hunk 2]
```diff
@@ -117,11 +117,15 @@ export const VeggieBurger: React.FC<{
     }
 
     return (
+        // TODO: Refactor this compone to work better without js. Where better
```
Comment (gtrufitt) [inline src/web/components/Nav/ExpandedMenu/ExpandedMenuToggle/VeggieBurger.tsx]:
```suggestion
        // TODO: Refactor this component to work better without js. Where better
```

**Comment by gtrufitt** [comment]
tenor-108031513

File: src/app/logging.ts (hunk 1 above)
Comment (oliverlloyd) [reply src/app/logging.ts]: 🤷‍♂ The missing no-undef rule bugged me for ages - I fully expect that rule to always be present; my brain gets freaked out when it's not

**Pull request merged**